v_frame/src/plane.rs

```diff
@@ -837,6 +921,127 @@ pub mod test {
     let v: Vec<_> = downsampled.iter().collect();
     assert_eq!(&expected[..], &v[..]);
   }
+
+  #[test]
+  fn test_plane_downscale() {
+    #[rustfmt::skip]
+    let plane = Plane::<u8> {
+      data: PlaneData::from_slice(&[
+        0, 0, 0, 0, 0, 0, 0, 0,
+        0, 0, 0, 0, 0, 0, 0, 0,
+        0, 0, 0, 0, 0, 0, 0, 0,
+        0, 0, 0, 1, 4, 5, 0, 0,
+        0, 0, 2, 3, 6, 7, 0, 0,
+        0, 0, 8, 9, 7, 5, 0, 0,
+        0, 0, 9, 8, 3, 1, 0, 0,
+        0, 0, 0, 0, 0, 0, 0, 0,
+        0, 0, 0, 0, 0, 0, 0, 0,
+      ]),
+      cfg: PlaneConfig {
+        stride: 8,
+        alloc_height: 9,
+        width: 4,
+        height: 4,
+        xdec: 0,
+        ydec: 0,
+        xpad: 0,
+        ypad: 0,
+        xorigin: 2,
+        yorigin: 3,
+      },
+    };
+    let downsampled = plane.downscale(2);
+
+    #[rustfmt::skip]
+    assert_eq!(
```

Comment:
You should compare the actual data and not the data + the padding.

Reply:
Fixed